react-carousel-component/my-app/src/CarouselComponent.tsx

```diff
@@ -0,0 +1,75 @@
+import { CSSProperties, useEffect, useState } from 'react';
+import { FaAngleRight, FaAngleLeft } from 'react-icons/fa';
+type Props = {
+  items: string[];
+};
+
+export function CarouselComponent({ items }: Props) {
+  const [current, setCurrent] = useState(0);
+  function handlePrevClick() {
+    setCurrent((current - 1 + items.length) % items.length);
+  }
+  function handleNextClick() {
+    setCurrent((current + 1) % items.length);
+  }
+  function handleSelect(index: number) {
+    setCurrent(index);
+  }
+
+  useEffect(() => {
+    const id = setTimeout(() => setCurrent((current + 1) % items.length), 1000);
+    return () => clearTimeout(id);
+  });
```

Comment:
All `useEffect`s should have a dependency array! It's possible you'll introduce infinite rerenders if you make `current` a dependency. If that happens, you can fix it in one of two ways:
1. Use `setCurrent((prev) => (prev + 1) % items.length)`
2. Use `useCallback` on `handleNextClick`, call it instead of `setCurrent`, and add it to the dependency array.